pkg/kubectl/cmd/util/factory.go

```diff
@@ -233,22 +306,42 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			return clients.ClientConfigForVersion(nil)
 		},
 		ClientForMapping: func(mapping *meta.RESTMapping) (resource.RESTClient, error) {
+			gvk, err := mapper.KindFor(mapping.GroupVersionKind.GroupVersion().WithResource(mapping.Resource))
```

Comment:
Is this a rebase slip or a intentional change? I remember I removed these lines months ago. You should use `mapping.GroupVersionKind` rather than calling `KindFor`.